### ISSUE TITLE
Fix create-obj / create-exe interaction

### ISSUE THREAD

**Issue by fschutt** [wasmerio/wasmer#3392]
Currently `create-obj` just creates a `static_defs.h` and a `wasm.h` file in the current directory whereas `create-exe` consumes this file. These tests are reliant on the current directory, so they break very easily if you run them from any other directory.

Currently this test will fail:

```
make build-wasmer
WASMER_DIR=./package 
    cargo llvm-cov --workspace --lcov 
    --output-path lcov.info 
    --features cranelift,singlepass,wasmer-artifact-create,static-artifact-create,wasmer-artifact-load,static-artifact-load,webc_runner 
    --exclude wasmer-integration-tests-ios
```

... because the `static_defs.h` is being generated, but not included in the `create-exe`. We need to find a way to make the interaction between `static_defs.h` and `wasm.h` as smooth as possible and also ideally not generate them in the current directory (since that might not be what the user wants). Also, we need to make tests more reliable by never assuming relative file paths, always canonicalize them and never rely on the current directory being stable (no `./blah.h`)

For example, this file path needs to be fixed to be configurable:

https://github.com/wasmerio/wasmer/blob/118b49efe727bb26099b2f6510f5fa7f4f76d442/lib/cli/src/commands/create_exe.rs#L241

... so that we can have an interaction between `create-obj` generating that `static-defs.h` file and `create-exe` consuming it.

**Comment by fschutt** [comment]
Tracking in https://github.com/wasmerio/wasmer/pull/3393

**Comment by syrusakbary** [comment]
This was fixed by #3531
